Hamburger menu pop up

kennankole opened the pull request:
**Implemented the following features using css and javascript:**

 - hamburger button mobile menu appears on click.
 - On clicking the (X) button, the mobile menu disappears.
 - On clicking mobile menu options, the mobile menu disappears.
 - On clicking the mobile menu options, a correct part of the page is displayed.

kennankole committed:
Added css and Js to transform humberger menu on click.

kennankole committed:
Fixing linters

Commit:
add funtion to menu button

kennankole committed:
Implemented menu items dissapear on click.

kennankole committed:
modified responsive.css file for mobile-version styles.

kennankole committed:
Fixed linter errors.

leonardodiasb commented on app/static/landing-page.css:
```diff
@@ -57,10 +57,12 @@ header {
 .right a:hover {
-  background-color: #6070ff;
+  /* background-color: #6070ff;
   color: #fff;
-  border-radius: 5px;
+  border-radius: 5px; */
```
- [ ] Please do not comment code. This can make your code messy. Take a look at the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

leonardodiasb commented on app/static/responsive.css:
```diff
@@ -35,6 +35,27 @@
+  }
+
+  .show-menu a {
+    font-size: 800;
+    font-size: 20px;
```
- [ ] Please do not use double attributes in your CSS file and try to use the correct attributes provided by the Figma template(remember to import all the font-weights used in the project). Take a look at this:
<img width="418" alt="image" src="https://user-images.githubusercontent.com/61919617/189721543-03357e1c-ab43-44a8-b453-ac1834b8f3d0.png">

leonardodiasb commented on app/static/js/index.js:
```diff
@@ -0,0 +1,14 @@
+function myFunction(x){
+    let show = document.querySelector('.right');
+    x.classList.toggle("change");
+    show.classList.toggle('show-menu');
+    show.classList.toggle('show-menu>a');
+}
+
+const humberger = document.querySelector(".hambuger-menu");
+const navMenu = document.querySelector(".right");
+
+document.querySelectorAll(".right").forEach(e => e.addEventListener("click", () => {
+    humberger.classList.remove("change");
+    navMenu.classList.remove("show-menu");
+}));
```
- [ ] Your mobile menu is working almost fine, but it's possible to close the menu if the user clicks on the blank space below the menu items. The user can accidentally click and close the menu this way and it's not the expected behavior. Please fix this issue.
 -
<img width="700" alt="image" src="https://user-images.githubusercontent.com/61919617/189723005-8816a639-11e8-45d1-8694-08c16f071c1d.png">

kennankole replied:
Hello @leonardodiasb 
Let I have worked on the changes you suggested.
Thanks.

leonardodiasb commented on app/static/responsive.css:
```diff
@@ -35,6 +35,27 @@
     display: none;
   }
 
+  .show-menu {
+    display: block;
+    position: absolute;
+    top: 85px;
+    line-height: 58px;
+    width: 335px;
+    z-index: auto;
+    height: 100%;
+    background-color: #6070ff;
+    margin-top: -85px;
+    text-align: start;
+    opacity: 0.8;
+  }
```
- [ ] Your menu looks quite similar to the Figma template, but the space from the first menu option to the top is not the same as the Figma template. Please add the correct space above the options.

leonardodiasb reviewed (changes requested):
# Required Changes ♻️ ⚠️ 🚦 

Hi @kennankole, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

![mario_ladders](https://user-images.githubusercontent.com/20563252/121369265-4e418880-c901-11eb-848c-1eff06d5a4cf.gif)

## Good points 👍 

 - No linter issues. ✔️
 - Professional PR. ✔️ 

## Aspects to improve ♻️

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

kennankole committed:
Fixed popup humbeger mobile menu
- Window closes only when user clicks on menu links.
- removed all commented blocks of code

HAMMAS-SALEEM commented on app/static/responsive.css:
```diff
@@ -35,6 +36,33 @@
     display: none;
   }
 
+  .show-menu {
+    display: block;
+    position: absolute;
+    top: 85px;
+    width: 335px;
+    z-index: auto;
+    height: 100%;
+    background-color: #6070ff;
+    margin-top: -85px;
+    text-align: start;
+    opacity: 0.8;
+  }
```
**[Optional]**
- [ ] You forgot to add the backdrop filter to blur your menu. Kindly add it to make it similar to the Figma template 😊

HAMMAS-SALEEM reviewed (approved):
# Status: Approved :green_circle:
![success](https://camo.githubusercontent.com/6ada59f6189497c1c2cae6d06f8154b7307a58643f232340e288140ed469eb28/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966)

## Hey @kennankole :wave:

Great work on implementing the requirements. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

:heavy_check_mark: Following the GitJin Flow
:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
